### PR TITLE
Remove modification of 'quantity' for FixedBoundary.X

### PR DESCRIPTION
**Pull request by henrikt-ma**
This follows the style in Modelica.Media.Interfaces.PartialMedium.MassFlowRate and gives tools a chance of making sense of the quantity.

Similar to https://github.com/modelica/ModelicaStandardLibrary/pull/4100.

### CODE REVIEW

**Comment by HansOlsson** [comment]
As in #4100 another option would be to drop the quantity.
The reasons are:

- We don't how to use it.
- Consistency: The PartialMedium has lots of massfractions - without any quantity.

File: Modelica/Fluid/Sources.mo [hunk 1]
```diff
@@ -29,7 +29,7 @@ package Sources "Define fixed or prescribed boundary conditions"
       annotation (Dialog(group="Boundary temperature or boundary specific enthalpy",
                   enable = not use_T));
     parameter Medium.MassFraction X[Medium.nX](
-         quantity=Medium.substanceNames) = Medium.X_default
+         quantity="MassFraction." + Medium.substanceNames) = Medium.X_default
```
Comment (henrikt-ma) [inline Modelica/Fluid/Sources.mo:32]: Trying to concretize @HansOlsson's alternative (but GitHub won't let me include the line about in the suggestion, making it look silly):
```suggestion
         ) = Medium.X_default
```

Comment (henrikt-ma) [reply Modelica/Fluid/Sources.mo:32]: Applying this change to keep this PR aligned with #4100.

**Comment by henrikt-ma** [comment]
Related to this, any opinions on this way of setting `quantity`?
```
    parameter Medium.ExtraProperty C[Medium.nC](
         quantity=Medium.extraPropertiesNames) = Medium.C_default
```
It looks like a problem similar to the one of this PR in that some sort of generic "name" is getting reused as `quantity`.  Would it make sense to open a separate issue/PR for this?

**Review by hubertus65 — APPROVED** [review]

**Review by beutlich — COMMENTED** [review]
PR title does not reflect the actual change.

**Comment by henrikt-ma** [comment]
> PR title does not reflect the actual change.

Thanks. Fixed.

**Comment by henrikt-ma** [comment]
Time to merge?

**Pull request merged**